autogen/agentchat/multimodal_conversable_agent.py

```diff
@@ -0,0 +1,118 @@
+from typing import Callable, Dict, List, Optional, Tuple, Union
+
+from autogen import oai
+from autogen.img_utils import gpt4v_formatter
+
+from .agent import Agent
+from .conversable_agent import ConversableAgent
+
+try:
+    from termcolor import colored
+except ImportError:
+
+    def colored(x, *args, **kwargs):
+        return x
+
+
+DEFAULT_LMM_SYS_MSG = """You are a helpful AI assistant.
+You can also view images, where the "<image i>" represent the i-th image you received."""
+
+
+class MultimodalConversableAgent(ConversableAgent):
+    def __init__(
+        self,
+        name: str,
+        system_message: Optional[Tuple[str, List]] = DEFAULT_LMM_SYS_MSG,
+        is_termination_msg=None,
+        *args,
+        **kwargs,
+    ):
+        """
+        Args:
+            name (str): agent name.
+            system_message (str): system message for the ChatCompletion inference.
+                Please override this attribute if you want to reprogram the agent.
+            **kwargs (dict): Please refer to other kwargs in
+                [ConversableAgent](conversable_agent#__init__).
+        """
+        super().__init__(
+            name,
+            system_message,
+            is_termination_msg=is_termination_msg,
+            *args,
+            **kwargs,
+        )
+
+        self.update_system_message(system_message)
```

Comment:
a quick question why we need to override sys msg here? isn't it updated once we pass it to the `super()._init_()` function?

Reply:
Because the LMM agent have "system_messge" as an array, handled by the update_system_message function. In the standard conversable agent, it is a string.

Reply:
Got it